### PR TITLE
Ume conversion/interaction tokenizer + fix SMILES and nucleotide tokenizers

### PR DESCRIPTION
**Pull request by karinazad**
* add two special tokens: `<convert>` and `<interact>` for later stages of Ume training:
* will be used as this: (or something like that)
`[CLS]  PROT_SEQ  [SEP] <convert> PROT_STRUCT(masked)  [SEP]` 
`[CLS]  PROT_SEQ  [SEP] <interact> SMILES(masked)  [SEP] `
* extend functionality of `UmeTokenizerTransform` to handle dual modalities
* change the name of `Ume` embedding method and allow embedding from existing input_ids

fix existing tokenizers:
* add lowercase normalized to nucleotide tokenizer (OG2 dataset contains a mix of upper and lowercase letters)
* `BPE` handled SMILES tokenization incorrectly, switch to WordLevel

### CODE REVIEW

File: src/lobster/tokenization/_ume_tokenizers.py [hunk 1]
```diff
@@ -76,17 +88,6 @@
 MASK_TOKEN = "<mask>"
 SPECIAL_TOKENS = {CLS_TOKEN, EOS_TOKEN, UNK_TOKEN, PAD_TOKEN, SEP_TOKEN, MASK_TOKEN}
 
-# Order of tokenizers, important for vocabulary construction
-# DO NOT change the order of tokenizers without making corresponding
-# changes to the `_add_reserved_tokens` func
-TOKENIZER_ORDER = [
```
Comment (kleinhenz) [inline src/lobster/tokenization/_ume_tokenizers.py:82]: What happened to this? Does it not matter anymore?

File: src/lobster/model/_ume.py [hunk 2]
```diff
@@ -292,12 +292,52 @@ def unfreeze(self) -> None:
...
+        aggregate : bool, default=True
+            Whether to average pool over the sequence length dimension.
+
+        Returns
+        -------
+        Tensor
+            Tensor of embeddings. If aggregate=True, shape is (batch_size, hidden_size).
+            Otherwise, shape is (batch_size, seq_len, hidden_size).
+        """
+        if not (k not in inputs for k in {"input_ids", "attention_mask"}):
```
Comment (kleinhenz) [inline src/lobster/model/_ume.py:311]: You could be more explicit here with `any` or `all`. I think you require `input_ids` but not `attention_mask` but I'm not sure.

File: src/lobster/tokenization/_ume_tokenizers.py [hunk 3]
```diff
@@ -529,3 +572,129 @@ def forward(
...
+        self.pad_id: int = self.tokenizer1.pad_token_id
+
+        if self.task_token not in self.tokenizer1.get_vocab() or self.task_token not in self.tokenizer2.get_vocab():
+            raise ValueError(
+                f"Task token '{self.task_token}' not found in tokenizer vocabularies. "
+                "Please ensure that the tokenizers are properly initialized."
+            )
+        self.task_token_id: int = self.tokenizer1.convert_tokens_to_ids(self.task_token)
+
+    def _encode_single_item(self, item: str | list[str], tokenizer, modality: Modality) -> dict:
```
Comment (kleinhenz) [inline src/lobster/tokenization/_ume_tokenizers.py:635]: Do you have a test to make sure this handles both `str` and `list[str]`?

File: src/lobster/tokenization/_ume_tokenizers.py [hunk 4]
```diff
@@ -461,43 +463,83 @@ def __init__(self):
...
+            - 'convert': For conversion tasks from one modality to another
+
+        Raises
+        ------
+        ValueError
+            If the `mode` is invalid or the special task token is missing in the tokenizer vocabulary.
+
+        Examples
+        --------
+        >>> from lobster.tokenization._ume_tokenizers import Ume2ModTokenizerTransform
```
Comment (kleinhenz) [inline src/lobster/tokenization/_ume_tokenizers.py:521]: class name is still `UmeTokenizerTransform`. Maybe makes sense to just keep it. It's not immediately obvious what 2Mod means imho

File: src/lobster/tokenization/_ume_tokenizers.py (hunk 1 above)
Comment (karinazad) [reply src/lobster/tokenization/_ume_tokenizers.py:82]: I defined this before but never actually used anywhere in the code

File: src/lobster/model/_ume.py (hunk 2 above)
Comment (karinazad) [reply src/lobster/model/_ume.py:311]: good point I'm missing `all`, it does need attention mask in FlexBERT (though not really since it later passes them to FlexBertModel which has attention mask optional but won't be changing that part of the code for now)

File: src/lobster/tokenization/_ume_tokenizers.py (hunk 4 above)
Comment (karinazad) [reply src/lobster/tokenization/_ume_tokenizers.py]: deleted this class and moved it all under `UmeTokenizerTransform` so it now handles single modality inputs and combos

File: src/lobster/tokenization/_ume_tokenizers.py (hunk 3 above)
Comment (karinazad) [reply src/lobster/tokenization/_ume_tokenizers.py]: after some trials and errors, decided to only support list of items for single modalities but for dual modalities, it only handles tuples e.g. `("VYWKL", "actg")`. This is mostly because 3d tokenizer returns a list of 4 strings for every structure and that gets really complicated when it's batched and would lead to a lot of silent mistakes I think

File: src/lobster/tokenization/_ume_tokenizers.py (hunk 4 above)
Comment (karinazad) [reply src/lobster/tokenization/_ume_tokenizers.py]:
```
tokenizer = UmeTokenizerTransform(
    modality = ("amino_acid", "SMILES"),
    max_length=16,
    mode="interact",
    return_modality=True,
)
```

File: src/lobster/tokenization/_ume_tokenizers.py [hunk 5]
```diff
@@ -461,71 +463,266 @@ def __init__(self):
...
+        Tokenizes inputs using the specified modality tokenizer
+        with vocabulary that's aware of reserved tokens.
+
+    For dual modality:
+        Combines inputs as: [CLS] [input1] [SEP] <task_token> [input2] [SEP]
+        where task_token is either <interact> or <convert>.
+
+    Examples:
+    ```
+    tokenizer = UmeTokenizerTransform(
```
Comment (kleinhenz) [inline src/lobster/tokenization/_ume_tokenizers.py:496]: nit: for these kinds of inline code examples you might as well use follow the [doctest](https://docs.python.org/3/library/doctest.html) conventions. I think we already do this in the main `Ume` class. Probably worth adding these to ci at some point

**Review by kleinhenz — APPROVED** [review]
